### ISSUE TITLE
A attempt to write finetuning script for codegen

### ISSUE THREAD

**Issue by vishalsingha** [salesforce/CodeGen#15]
Hii, 
I am interested in writing finetuning script for this model.

Can anyone tell while training the model in what format can i provide output to the model.

As the data is of form
{code:"def hello(name):    return f"Hello {name}", nl : "This function takes name as input and a massage saying hello to the person in format hello name."}

We can tokenize input (either code or text )  by the making tokenizer and passing text into it.

But what format should I use for training and how to compare loss of original output and predicted output.

Thanks

**Comment by vishalsingha** [comment]
Hello, 
Can anyone give me the small sample of BIGPYTHON dataset used for training the Codegen-nl mono.

**Comment by nashid** [comment]
@vishalsingha wondering whether you have made any progress in this regard?

**Comment by vishalsingha** [comment]
> @vishalsingha wondering whether you have made any progress in this regard?

Yes I have used the training script of CodeT5 model for training this network after slight modifications.

**Comment by nashid** [comment]
@vishalsingha do you plan to share it?

**Comment by nashid** [comment]
@vishalsingha got it. Would be awesome if you share any insight how you approached developing the fine tuning script. Also any reference to open source resources would be much appreciated.
